### PR TITLE
Add Post

### PR DESCRIPTION
**Pull request by funnyboy-roks**

### CODE REVIEW

**Comment by funnyboy-roks** [comment]
Want to get this merged so that we reduce unnecessary conflicts

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -12,11 +12,13 @@ concurrency:
 name: build
 jobs:
     build:
-        runs-on: ubuntu-latest
-        name: build
         env:
             # This is a fake connection string that is placed just for the compiler to stop whining
             POSTGRES_CONNECTION_STRING: 'postgresql://username:password@host/db_name'
+            PUBLIC_PREVIEW_HOST: ${{ secrets.PUBLIC_PREVIEW_HOST }}
+            PUBLIC_LIBRARY_ID: ${{ secrets.PUBLIC_LIBRARY_ID }}
```
Comment (jackjohn7) [inline .github/workflows/build.yml:19]: I don't think this particular variable exists. Is the purpose of having actual values (as opposed to dummy ones) here for running tests interacting with video in some sort of "test" library.

File: package.json [hunk 2]
```diff
@@ -58,6 +58,7 @@
         "postgres": "^3.4.4",
         "tailwind-merge": "^2.5.2",
         "tailwind-variants": "^0.2.1",
+        "vaul-svelte": "^0.3.2",
```
Comment (jackjohn7) [inline package.json:61]: What's the advantage of this drawer over the shadcn svelte one? 
https://www.shadcn-svelte.com/docs/components/drawer

File: src/lib/components/Post.svelte [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    import ArrowUp from 'lucide-svelte/icons/arrow-up';
+    import Share2 from 'lucide-svelte/icons/share-2';
+    import Play from 'lucide-svelte/icons/play';
+
+    import Score from './Score.svelte';
+    import ShareSheet from './ShareSheet.svelte';
+    import UserChannel from './UserChannel.svelte';
+    import Player from './Player.svelte';
+
+    const videoId = 'e0245338-7c04-4a6c-b44f-0e279a849cf5';
```
Comment (jackjohn7) [inline src/lib/components/Post.svelte:21]: This should be parameterized, no?

File: src/lib/components/Post.svelte [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            downvote_pressed = false;
+            new_state = !upvote_pressed;
+        } else {
+            upvote_pressed = false;
+            new_state = !downvote_pressed;
+        }
+        console.log(dir, new_state);
+    };
+
+    let playing_video = false;
```
Comment (jackjohn7) [inline src/lib/components/Post.svelte:41]: might be useful later for this to be a bindable.

File: src/lib/components/Post.svelte [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    const vote = (dir: 'up' | 'down') => {
+        let new_state;
+        if (dir === 'up') {
+            downvote_pressed = false;
+            new_state = !upvote_pressed;
+        } else {
+            upvote_pressed = false;
+            new_state = !downvote_pressed;
+        }
+        console.log(dir, new_state);
```
Comment (jackjohn7) [inline src/lib/components/Post.svelte:38]: probably want to dispatch an event here

File: src/routes/(app)/+page.svelte [hunk 6]
```diff
@@ -20,3 +21,5 @@
         >the Wiki</a
     > to read the documentation
 </p>
+
+<Post />
```
Comment (jackjohn7) [inline src/routes/(app)/+page.svelte:25]: I know we're wanting to display that squirrel video, so I'd rather we hardcode that here than within the component

File: tailwind.config.ts [hunk 7]
```diff
@@ -15,6 +16,8 @@ const config: Config = {
         },
         extend: {
             colors: {
+                upvote: colors.orange[600],
+                downvote: colors.cyan[600],
```
Comment (jackjohn7) [inline tailwind.config.ts:20]: Love this!

**Review by jackjohn7 — CHANGES_REQUESTED** [review]
Looks great! The component scales horizontally in a way I think is a bit unnatural, but I understand we can't really assess that until we have more UI elements. Just a few changes and I think this is good to be merged!

File: tailwind.config.ts (hunk 7 above)
Comment (jackjohn7) [reply tailwind.config.ts:20]: Feel free to mark this resolved

File: package.json (hunk 2 above)
Comment (funnyboy-roks) [reply package.json:61]: Shadcn depends upon that library for the drawer.

> Drawer is built on top of [Vaul Svelte](https://vaul-svelte.com/), which is a Svelte port of [Vaul](https://vaul.emilkowal.ski/) by [Emil Kowalski](https://twitter.com/emilkowalski_).

File: src/lib/components/Post.svelte (hunk 3 above)
Comment (funnyboy-roks) [reply src/lib/components/Post.svelte:21]: You're right.  My intention was to parameterise it when working on the infinite scroll component, same with the user and other information about the post itself.

File: src/lib/components/Post.svelte (hunk 4 above)
Comment (funnyboy-roks) [reply src/lib/components/Post.svelte:41]: I agree, will do that.

File: src/lib/components/Post.svelte (hunk 5 above)
Comment (funnyboy-roks) [reply src/lib/components/Post.svelte:38]: I agree.  My intention was to add all binding and external interactions when I make the infinite scroll, but I'll go ahead and do this now.

File: .github/workflows/build.yml (hunk 1 above)
Comment (funnyboy-roks) [reply .github/workflows/build.yml:19]: The host exists, I forgot to add the library id, but thinking about it, there's no reason these couldn't be dummy values, I'm just in the habit of supplying real values.

Comment (jackjohn7) [reply .github/workflows/build.yml:19]: Totally fair. I think this approach may be useful if we end up having any tests relating to video services

Comment (funnyboy-roks) [reply .github/workflows/build.yml]: these have been replaced by dummy variables

File: src/lib/components/Post.svelte (hunk 3 above)
Comment (funnyboy-roks) [reply src/lib/components/Post.svelte]: The Post component has been fully parameterised.

File: src/lib/components/Post.svelte (hunk 5 above)
Comment (funnyboy-roks) [reply src/lib/components/Post.svelte]: Added in latest commit.

File: src/routes/(app)/+page.svelte (hunk 6 above)
Comment (funnyboy-roks) [reply src/routes/(app)/+page.svelte:36]: Also in latest commit.

File: src/lib/components/Post.svelte (hunk 4 above)
Comment (funnyboy-roks) [reply src/lib/components/Post.svelte]: Done in latest commit.

**Review by jackjohn7 — APPROVED** [review]
Looks great!

**Pull request merged**